lib/validator.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+const P        = require('bluebird');
+const Ajv      = require('ajv');
+
+const urlGetObject = require('./event-utils').urlGetObject;
+
+
+// TODO Add debug/trace logging of schemas as they are cached and used.
+
+/**
+ * Caches AJV validators by schema urls.
+ *
+ * Usage:
+ *  validatorCache = new ValidatorCache();
+ *  validator = validatorCache.validatorFor('file:///path/to/schema.yaml');
+ *  schema = validatorCache.schemaAt('file:///path/to/schema.yaml');
+ */
+class ValidatorCache {
```

Comment:
I'm not entirely sure we need this. You can add schemas directly to `AJV` object with an added benefit that it supports references.

Reply:
Ya we might not need to cache the validator itself, but I did need a cached mapping from schema_url to either scheme or validator in order to avoid extra URL gets, and caching the validator seemed easier.  We can reevaluate this for sure.